static/js/components/inputs/SelectField.js

```diff
@@ -13,169 +14,73 @@ import type {
-    // the selected option label if one is selected, or an empty string
     const {
+      updateValidationVisibility,
+      validator,
+      updateProfile,
       profile,
       keySet,
-      updateProfile,
-      validator,
-      updateValidationVisibility
     } = this.props;
-    let clone = _.cloneDeep(profile);
-    _.set(clone, this.editKeySet, undefined);
     updateValidationVisibility(keySet);
-    updateProfile(clone, validator);
+    updateProfile(profile, validator);
   };
 
-  onNewRequest: Function = (optionOrString: Option|string, index: number): void => {
-    const {
-      profile,
-      updateProfile,
-      keySet,
-      options,
-      onChange,
-      validator,
-    } = this.props;
-    let clone = _.cloneDeep(profile);
-    let toStore;
-    if (index === -1) {
-      // enter was pressed and optionOrString is a string
-      // select first item in dropdown if any are present
-      let autocompleteProps = this.getAutocompleteProps();
-      let filterFunction = _.has(autocompleteProps, 'filter') ?
-        autocompleteProps.filter :
-        defaultFilter;
-      let firstMatchingOption = options.find((option) => filterFunction(optionOrString, option.label));
-      if (firstMatchingOption) {
-        toStore = firstMatchingOption.value;
-      }
-    } else {
-      // user selected an item in the menu
-      if ( typeof optionOrString !== 'string' ) {
-        toStore = optionOrString.value;
-      }
-    }
-
-    if (toStore !== undefined) {
-      _.set(clone, keySet, toStore);
-    } // else we couldn't figure out what the user wanted to select, so leave it as is
-    _.set(clone, this.editKeySet, undefined);
+  label: Function = (label: string): React$Element<*> => (
+    <div className="select-label">
+      { label }
+    </div>
+  );
 
-    updateProfile(clone, validator);
-    if (_.isFunction(onChange)) {
-      onChange(clone);
-    }
-  };
-
-  getAutocompleteProps: Function = (): Object => {
-    const {
-      autocompleteBehaviors,
-      autocompleteStyleProps,
-      label,
-      options,
-      keySet,
-      errors
-    } = this.props;
-    // Call the autocompleteBehaviors array in series and combine their results into a single object to pass as props
-    let autocompleteProps = Object.assign({}, ...callFunctionArray(autocompleteBehaviors, this.props));
-    return Object.assign({}, {
-      ref: "autocomplete",
-      animated: false,
-      menuCloseDelay: 0,
-      dataSource: options,
-      floatingLabelText: label,
-      onNewRequest: this.onNewRequest,
-      onUpdateInput: this.onUpdateInput,
-      onBlur: this.onBlur,
-      errorText: _.get(errors, keySet)
-    }, autocompleteStyleProps, autocompleteProps);
+  className = (): string => {
+    const { className, label } = this.props;
+    const ifNil = R.compose(_.snakeCase, R.defaultTo(""));
+    return `select-field ${ifNil(className)} ${ifNil(label)}`;
```

Comment:
Oh yes, I forgot - this isn't receiving camelCase *per se*.

Basically i'm taking advantage of the fact that:

```js
_.snakeCase('Some Words') == 'some_words';
```

to remove spaces and turn a potentially multi-word label into something more sensible (a label like `'Preferred Language'`, for instance, becomes `'preferred_language'`).